site/content/docs/01-structure.md

```diff
@@ -0,0 +1,112 @@
+
+If you take a look inside the [sapper-template](https://github.com/sveltejs/sapper-template) repo, you'll see some files that Sapper expects to find:
+
+```bash
+├ package.json
+├ src
+│ ├ routes
+│ │ ├ # your routes here
+│ │ ├ _error.svelte
+│ │ └ index.svelte
+│ ├ client.js
+│ ├ server.js
+│ ├ service-worker.js
+│ └ template.html
+├ static
+│ ├ # your files here
+└ rollup.config.js / webpack.config.js
+```
+
+When you first run Sapper, it will create an additional `__sapper__` directory containing generated files.
+
+You'll notice a few extra files and a `cypress` directory which relates to [testing](docs#testing) — we don't need to worry about those right now.
+
+> You *can* create these files from scratch, but it's much better to use the template. See [getting started](docs#getting-started) for instructions on how to easily clone it
+
+
+### package.json
+
+Your package.json contains your app's dependencies and defines a number of scripts:
+
+* `npm run dev` — start the app in development mode, and watch source files for changes
+* `npm run build` — build the app in production mode
+* `npm run export` — bake out a static version, if applicable (see [exporting](docs#exporting))
+* `npm start` — start the app in production mode after you've built it
+* `npm test` — run the tests (see [testing](docs#testing))
+
+
+### src
+
+This contains the three *entry points* for your app — `src/client.js`, `src/server.js` and (optionally) `src/service-worker.js` — along with a `src/template.html` file.
+
+#### src/client.js
+
+This *must* import, and call, the `start` function from the generated `@sapper/app` module:
+
+```js
+import * as sapper from '@sapper/app';
+
+sapper.start({
+	target: document.querySelector('#sapper')
+});
+```
+
+In many cases, that's the entirety of your entry module, though you can do as much or as little here as you wish. See the [client API](docs#client-api) section for more information on functions you can import.
+
+
+#### src/server.js
+
+This is a normal Express (or [Polka](https://github.com/lukeed/polka), etc) app, with three requirements:
+
+* it should serve the contents of the `static` folder, using for example [sirv](https://github.com/lukeed/sirv)
+* it should call `app.use(sapper.middleware())` at the end, where `sapper` is imported from `@sapper/server`
+* it must listen on `process.env.PORT`
+
+Beyond that, you can write the server however you like.
+
+
+#### src/service-worker.js
+
+Service workers act as proxy servers that give you fine-grained control over how to respond to network requests. For example, when the browser requests `/goats.jpg`, the service worker can respond with a file it previously cached, or it can pass the request on to the server, or it could even respond with something completely different, such as a picture of llamas.
```

Comment:
I don't know whether I was supposed to think about goatse when I read goats.jpg but I did

Reply:
The young 'uns will never know the horrors we internet old-timers have seen